From 92b3b0cf1bb937eb3a175fbe6e8219513bc16876 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 21 Nov 2022 09:03:03 +0100
Subject: [PATCH 0827/1010] evl/mutex: remove invalid assertion

We may enter adjust_owner_boost() with no boosters because we did not
hold owner->lock on entry, therefore this owner might have dropped all
of them in the meantime if not current on the local CPU.

Drop the assertion marking this condition as an issue. However, check
that T_BOOST is cleared if the latter is met.

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/mutex.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/kernel/evl/mutex.c b/kernel/evl/mutex.c
index 9cc0f141a7..3217d841de 100644
--- a/kernel/evl/mutex.c
+++ b/kernel/evl/mutex.c
@@ -253,7 +253,8 @@ static void adjust_owner_boost(struct evl_thread *owner)
 
 	raw_spin_lock(&owner->lock);
 
-	if (EVL_WARN_ON_ONCE(CORE, list_empty(&owner->boosters))) {
+	if (list_empty(&owner->boosters)) {
+		EVL_WARN_ON_ONCE(CORE, owner->state & T_BOOST);
 		raw_spin_unlock(&owner->lock);
 		return;
 	}
-- 
2.30.2

